### ISSUE TITLE
ImGuiKey_ deprecated?

### ISSUE THREAD

**Issue by passivedragon** [thedmd/imgui-node-editor#183]
I'm only just getting into IMGui and wanted to check out this project.
I've gotten a small project together with just IMGui that works fine, but adding imgui-node-editor into the mix completely fails with a series of similar-ish errors, one of which is `‘ImGuiKey_’ was not declared in this scope`.

After a quick look through the source, it seems that ImGuiKey_ may have been an enum type alias for int type ImGuiKey, which now is just `enum ImGuiKey : int`;. These errors get thrown each time `DECLARE_KEY_TESTER` is expanded.

**Comment by thedmd** [comment]
Please tell me what version of ImGui you're using?

**Comment by passivedragon** [comment]
I pulled the latest from master directly at the time, which ended up being https://github.com/ocornut/imgui/commit/a241dc7990b631fde6575771173c2442d43d2812
I do recall looking over the source when I had this issue, and I believe this change with ImGuiKey_ has been in there since... around 1.83 or so? (I'm probably wrong there)

Also seems like my latest comment didn't make it to github. I ended up fixing it on my end by taking a few small changes. I only realized after some time that this project isn't drop-in suitable in the way I was doing things. Would be nice if the references to imgui were easier to point to difference places when linking/building, for those including imgui on the side, along imgui-node-editor

**Comment by thedmd** [comment]
Support for extra keys was my extension to ImGui, some time later it landed in library itself.
That make code using `ImGuiKey_` is not longer needed. I will pinpoint ImGui version number and ifdef it out.
Node Editor work with old ImGui as well latest version. Latter need fixes sometimes, like in this case. Sorry I didn't catch that before you stumbled on the issue.

ImGui does have `IMGUI_VERSION_NUM` from some time. It is bumped when something non-trivial change.

> this project isn't drop-in suitable in the way I was doing things

Idea of integrating it into your code base is that you include sources from root directory only and they should work.
What made it unsuitable for your use case?

Can you tell me if define `IMGUI_DISABLE_OBSOLETE_KEYIO`?

**Comment by thedmd** [comment]
Support for latest ImGui was pushed to `develop` branch.

**Comment by passivedragon** [comment]
> > this project isn't drop-in suitable in the way I was doing things
> 
> Idea of integrating it into your code base is that you include sources from root directory only and they should work. What made it unsuitable for your use case?
> 
> Can you tell me if define `IMGUI_DISABLE_OBSOLETE_KEYIO`?

That was misunderstanding on my part, when I first added this to my project I grabbed some of the example files, but didn't like the external/imgui part, so when I then stumbled over includes pointing at `external/` it took me a bit to properly sort things out in my project structure.
I could see the includes just poniting at `imgui/imgui.h` and just supplying an `-Iexternal/` when building for the examples, but that's not really a big issue, I just didn't check thoroughly enough at first.

Thanks a ton for the quick response and fix! I'll be using the develop branch then.
